### ISSUE TITLE
RET505 False Positive

### ISSUE THREAD

**Issue by jasonwashburn** [charliermarsh/ruff#3055]
Seems like `RET505` has come up in issues a few times before, but I seem to be getting false positives as well. Here is a pretty simple example of a code snippet that provokes the error.

```python
def is_even(value: int) -> bool:
    """Check if a value is even."""
    if value % 2 == 0:
        print(f"{value} is even")
        return True
    else:
        print(f"{value} is odd")
        return False
```

When running ruff with the defaults (no config file) and adding select for RET505 produces the following:

```bash
❯ ruff --select=RET505 false_positive.py
false_positive.py:6:5: RET505 Unnecessary `else` after `return` statement
Found 1 error.
```

```bash
❯ ruff --version        
ruff 0.0.247
```

**Comment by charliermarsh** [comment]
I believe this is actually working as intended -- it wants you to do:

```py
def is_even(value: int) -> bool:
    """Check if a value is even."""
    if value % 2 == 0:
        print(f"{value} is even")
        return True

    print(f"{value} is odd")
    return False
```

Since that code is equivalent: if the number is even, it'll hit the `if`, then `return` at the end of the `if` block. So the `else` isn't necessary.

**Comment by jasonwashburn** [comment]
ugh...you're right...lol, it's so simple I looked right past that...that's what I get for validating/submitting an issue before my second cup of coffee...

**Comment by charliermarsh** [comment]
Haha no it’s all good! These rules can be hard to reason about.
